python/libraries/bimvee/visualiser.py

```diff
@@ -355,4 +355,128 @@ def get_dims(self):
         return self.renderX, self.renderY
 
     def get_colorfmt(self):
-        return 'rgb'
\ No newline at end of file
+        return 'rgb'
+
+class VisualiserPoint3(Visualiser):
+
+    renderX = 300 # TODO Hardcoded
+    renderY = 300
+    labels = None
+
+    def __init__(self, data):
+        self.set_data(data)
+        self.smallestRenderDim = min(self.renderX, self.renderY)
+
+    '''
+    Offset and scale the pose translations so that they all fit into the volume:
+        x [-0.5:0.5]
+        y[-0.5:0.5]
+        z[1:2]
+    '''
+    def set_data(self, data):
+        # scale and offset point data so that it remains proportional 
+        # but stays in the range 0-1 for all dimensions
+        pointX = data['point'][:, 0]
+        pointY = data['point'][:, 1]
+        pointZ = data['point'][:, 2]
+        minX = np.min(pointX)
+        maxX = np.max(pointX)
+        minY = np.min(pointY)
+        maxY = np.max(pointY)
+        minZ = np.min(pointZ)
+        maxZ = np.max(pointZ)
+        centreX = (minX + maxX) / 2
+        centreY = (minY + maxY) / 2
+        centreZ = (minZ + maxZ) / 2
+        largestDim = max(maxX-minX, maxY-minY, maxZ-minZ)
+        if largestDim == 0:
+            largestDim = 1
+
+        pointScaled = np.empty_like(data['point'])
+        pointScaled[:, 0] = pointX - centreX
+        pointScaled[:, 1] = pointY - centreY
+        pointScaled[:, 2] = pointZ - centreZ
+        pointScaled = pointScaled / largestDim + 0.5
+        internalData = {'ts': data['ts'],
+                        'point': pointScaled
+                        }
+        self.__data = internalData
+            
+    def project3dTo2d(self, x=0, y=0, z=0, smallestRenderDim=1, **kwargs):
+        if kwargs.get('perspective', True):
+            # Move z out by 0.5, so that the data is between 0.5 and 1.5 distant in z
+            x = (x - 0.5) / (z + 0.5) + 0.5
+            y = (y - 0.5) / (z + 0.5) + 0.5
+        x *= smallestRenderDim
+        y *= smallestRenderDim
+        x = x.astype(int)
+        y = y.astype(int)
+        return x, y
+
+    def project_point(self, point, image, **kwargs):
```

Comment:
```suggestion
    def project_pointcloud_on_image(self, point, image, **kwargs):
```
Can we refactor function name to make the functionality clearer?

Reply:
Refactored in new commit.